cloudify_hostpool/rest/backend.py

```diff
@@ -1,164 +1,370 @@
-########
-# Copyright (c) 2015 GigaSpaces Technologies Ltd. All rights reserved
+# #######
+# Copyright (c) 2016 GigaSpaces Technologies Ltd. All rights reserved
 #
 # Licensed under the Apache License, Version 2.0 (the "License");
 # you may not use this file except in compliance with the License.
 # You may obtain a copy of the License at
 #
-# http://www.apache.org/licenses/LICENSE-2.0
+#        http://www.apache.org/licenses/LICENSE-2.0
 #
 # Unless required by applicable law or agreed to in writing, software
 # distributed under the License is distributed on an "AS IS" BASIS,
-# * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-# * See the License for the specific language governing permissions and
-# * limitations under the License.
+#    * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    * See the License for the specific language governing permissions and
+#    * limitations under the License.
+'''
+    cloudify_hostpool.rest.backend
+    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    RESTful service backend methods
+'''
 
+from copy import deepcopy
+from collections import Mapping
 import filelock
-import uuid
-import os
+import logging
+import socket
 
-from cloudify_hostpool.hosts import scan
-from cloudify_hostpool.storage import sqlite
-from cloudify_hostpool.config import yaml_pool
-from cloudify_hostpool import exceptions
+# Used for IP / CIDR routines
+from netaddr import IPNetwork
+from netaddr.core import AddrFormatError
 
+from cloudify_hostpool import constants
+from cloudify_hostpool.storage.tinydb_nosql import Database
+from cloudify_hostpool import exceptions
 
 # we currently don't expose these in the configuration because its somewhat
 # internal. perhaps at a later time we can have this configurable, at which
 # point we need to define the semantics of how to initialize the components.
+FLOCK = filelock.FileLock('host-pool-backend.lock')
 
-FLock = filelock.FileLock('host-pool-backend.lock')
+# HostAlchemist
+# - Converts user-provided host entries into a consumable structure
+# HostReconciler
+# - Handles any host entry duplication cases
 
-# if this file exists, loading of the pool will not take place.
-# note that this means that multiple instances of this application that
-# execute from the same directory will only load the pool the first
-# time, and all other instances will use the previously loaded pool.
-INDICATOR = 'host-pool-loaded-indicator'
 
+class HostAlchemist(object):
```

Comment:
No, does it need to? I see no harm in allowing to be re-used later vs. having one giant, monolithic RestBackend class when the functionality differs.